sinabs/backend/dynapcnn/chips/dynapcnn.py

```diff
@@ -90,149 +92,169 @@ def get_dynapcnn_layer_config_dict(cls, layer: DynapcnnLayer):
-                chip_layer = config.dvs_layer
-                cls.write_dvs_layer_config(chip_equivalent_layer, chip_layer)
-                has_dvs_layer = True
-            elif isinstance(chip_equivalent_layer, DynapcnnLayer):
-                chip_layer = config.cnn_layers[chip_layers[i_cnn_layer]]
-                cls.write_dynapcnn_layer_config(chip_equivalent_layer, chip_layer)
-                i_cnn_layer += 1
-            else:
-                # in our generated network there is a spurious layer...
-                # should never happen
-                raise TypeError("Unexpected layer in the model")
+        Parameters
+        ----------
+        - model (DynapcnnNetwork): network instance used to read out `DynapcnnLayer` instances.
+
+        Returns
+        ----------
+        - config (DynapcnnConfiguration): an instance of a `DynapcnnConfiguration`.
+        """
+        config = cls.get_default_config()
 
-            if i == len(layers) - 1:
-                # last layer
-                chip_layer.destinations[0].enable = False
+        if type(model) == sinabs.backend.dynapcnn.dynapcnn_network.DynapcnnNetwork:
+            """ Loops through `DynapcnnNetworkGraph._layers_mapper`, containing all `DynapcnnLayer`s in the model, their
+            core ID (where they are configured onto) and their target destinations. Each `ith_dcnnl` has all the info. necessary to config.
+            their respective `CNNLayerConfig` object.
+            """
+            has_dvs_layer = False   # TODO DVSLayer not supported yet.
+
+            for layer_index, ith_dcnnl in model.layers_mapper.items():
+                if isinstance(ith_dcnnl, DVSLayer):
+                    # TODO DVSLayer not supported yet.
+                    pass
+
+                elif isinstance(ith_dcnnl, DynapcnnLayer):
+                    # retrieve assigned core from the handler of this DynapcnnLayer (`ith_dcnnl`) instance.
+                    chip_layer = config.cnn_layers[model.layers_handlers[layer_index].assigned_core]
+                    # write core configuration.
+                    cls.write_dynapcnn_layer_config(ith_dcnnl, chip_layer, model.layers_handlers[layer_index], model.layers_handlers)
+
+                else:
+                    # shouldn't happen since type checks are made previously.
+                    raise TypeError(f"Layer (index {layer_index}) is unexpected in the model: \n{ith_dcnnl}")
+                
+            if not has_dvs_layer:
+                # TODO DVSLayer not supported yet.
+                config.dvs_layer.pass_sensor_events = False
             else:
-                # Set destination layer
-                chip_layer.destinations[0].layer = chip_layers[i_cnn_layer]
-                chip_layer.destinations[0].enable = True
+                config.dvs_layer.pass_sensor_events = False
 
-        if not has_dvs_layer:
-            config.dvs_layer.pass_sensor_events = False
+        else:
```

Comment:
Implemented suggestion.
@Willian-Girao Please review and resolve if appropriate.

Reply:
Is this outdated maybe? It also seems that the comment in on the wrong lines. In the last version I pushed:

```
        if type(model) == sinabs.backend.dynapcnn.dynapcnn_network.DynapcnnNetwork:
            """ Loops through `DynapcnnNetworkGraph._layers_mapper`, containing all `DynapcnnLayer`s in the model, their
            core ID (where they are configured onto) and their target destinations. Each `ith_dcnnl` has all the info. necessary to config.
            their respective `CNNLayerConfig` object.
            """
            has_dvs_layer = False   # TODO DVSLayer not supported yet.

            for layer_index, ith_dcnnl in model.layers_mapper.items():
                if isinstance(ith_dcnnl, DVSLayer):
                    # TODO DVSLayer not supported yet.
                    pass

                elif isinstance(ith_dcnnl, DynapcnnLayer):
                    # retrieve assigned core from the handler of this DynapcnnLayer (`ith_dcnnl`) instance.
                    chip_layer = config.cnn_layers[model.layers_handlers[layer_index].assigned_core]
                    # write core configuration.
                    cls.write_dynapcnn_layer_config(ith_dcnnl, chip_layer, model.layers_handlers[layer_index], model.layers_handlers)

                else:
                    # shouldn't happen since type checks are made previously.
                    raise TypeError(f"Layer (index {layer_index}) is unexpected in the model: \n{ith_dcnnl}")
                
            if not has_dvs_layer:
                # TODO DVSLayer not supported yet.
                config.dvs_layer.pass_sensor_events = False
            else:
                config.dvs_layer.pass_sensor_events = False

        else:
            raise TypeError(f"Unexpected model {type(model)}.")
```

Reply:
Interesting, it doesn't seem like I have made any changes to this file.
Maybe my 2nd comment should have ended up elsewhere.
What I meant to do is:
```
if not isinstance(model, DynapcnnNetwork):
    raise TypeError(...)
```
and then the code block that until now is inside the 'if' statement.

Reply:
Okay, looks like I did that, then deleted that part again, because the type-check was a bit superfluous. Sorry for the confusion. I will close this conversation.